### DISCUSSION TITLE
Postgres access with ojs

### DISCUSSION THREAD

**Discussion by lyon-fnal** [quarto-dev/quarto-cli#3656] [Q&A]
Hi, I'm wondering if it is possible to access a Postgres database using ojs cells. Observable seems to have their own database access system - I don't think that works with Quarto. If I try to,

````
```{ojs}
require("postgres")
```
````

gives the dreaded `RequireError: invalid module`. I've also tried the `pq` package and get the same result. Tips at https://observablehq.com/@observablehq/how-to-require-stubborn-modules didn't help. 

Any ideas here? Being able to directly access a large database with an interactive ojs document would be a very nice feature to have. Thanks!

**Comment by cscheid** [comment]
I don't that's a quarto issue, since that module doesn't work on observablehq.com either:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/285675/207381872-88abe856-d105-4971-8aa6-d99ef9b0e138.png">

If you find a javascript module that can access a postgres database, then that should work. (Are postgres databases directly available through HTTP? It would appear that you need, at least, something like https://github.com/PostgREST/postgrest)

**Reply by lyon-fnal** [reply]
Thanks @cscheid -Observable seems to have their own DB connection system described at https://observablehq.com/@observablehq/databases . They do have a self-hosted proxy - maybe I'll try that. I think you're right that the `postgres` package won't work in this context - I think that's meant to run in `node` and not in the browser.

**Reply by cscheid** [reply]
Do note that that page is tagged with their `pro` label, which is, generally speaking, part of observablehq.com's proprietary offering that quarto does not have access to.

**Reply by lyon-fnal** [reply]
Indeed. It looks like `DatabaseClient` is not part of the public stdlib.
